src/puppetlabs/puppetdb/config.clj

```diff
@@ -25,34 +25,54 @@
 ;; support defaulting/converting nested maps, these configs can be put
 ;; together in a single schema that defines the config for PuppetDB
 
+(defn warn-unknown-keys
+  [schema data]
+  (doseq [k (pls/unknown-keys schema data)]
+    (log/warn (format "The configuration item `%s` does not exist and should be removed from the config." k))))
+
+(defn warn-and-validate
+  "Warns a user about unknown configurations items, removes them and validates the config."
+  [schema data]
+  (warn-unknown-keys schema data)
+  (let [sanitized-data (pls/strip-unknown-keys schema data)]
+    (s/validate schema sanitized-data)
+    sanitized-data))
```

Comment:
I think this line can be removed -- s/validate will return sanitized-data if it matches the schema.